src/librustc_mir/borrow_check/mod.rs

```diff
@@ -883,7 +883,7 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         // Check is_empty() first because it's the common case, and doing that
         // way we avoid the clone() call.
         if !self.access_place_error_reported.is_empty()
-            && self.access_place_error_reported.contains(&(place_span.0.clone(), place_span.1))
+            && self.access_place_error_reported.contains(&(*place_span.0, place_span.1))
```

Comment:
@oli-obk @spastorino Maybe we need to consider passing `Place` by value more often?
(it's a pair of an integer and a pointer, so it should be really compact)

Reply:
That doesn't work if there are subslices of the projection involved. We do have a PlaceRef type for exactly that use case though

Reply:
Yeah, except in the cases @oli-obk mentions I'd change most of the uses to take `Place` by value. I need to do a bunch of cleanups still :).